### PR TITLE
E_CLASSROOM-342 [FE/BE] Implement Change Password for Student Profile

### PR DESCRIPTION
**Pull request by ErickDoroy**
### Issue Link
- https://framgiaph.backlog.com/view/E_CLASSROOM-342
### Definition of Done
- [x] Able to Change password

## Related PR:
- https://github.com/framgia/sph-classroom-els-fe/pull/202
#### Main note
- Try logging in again after updating the password to test whether password was successfully updated.

- To test in Postman, use this route: http://localhost:82/api/v1/password-edit
  Required Fields: `password`, `new_password`, `password_confirmation`

#### Pages Affected
- N/A

#### Scenarios/ Test Cases
- [x] we should expected the password will change

### Screenshots
image
image

### CODE REVIEW

File: app/Http/Controllers/API/v1/Student/StudentController.php [hunk 1]
```diff
@@ -57,4 +59,18 @@ public function show(Request $request)
...
+        if(!Hash::check($request->password, $student->password)){
+            return $this->errorResponse(['password' =>  trans('auth.password')], 401); 
+        }
+
+        $student->password = bcrypt($request->new_password);
+        $student->save();
+
+        return $this->successResponse(['message' => 'Your password has been successfully updated.'], 200);
+    }
 }
```
Comment (karlolee50) [inline app/Http/Controllers/API/v1/Student/StudentController.php]: add extra line

File: routes/api.php [hunk 2]
```diff
@@ -69,6 +69,7 @@
         Route::get('/quiz_attempts/{quiz_id}', [UserScoresAndAttemptsController::class, 'UserScoreAndAttempts']);
         Route::post('/profileEdit', [ChangeNameEmailController::class, 'changeName']);
         Route::post('/profileEdituploadImage', [UploadImageController::class, 'uploadAvatar']);
+        Route::patch('/password-edit', [StudentController::class, 'updatePassword']);
```
Comment (karlolee50) [inline routes/api.php]:
```suggestion
        Route::patch('/edit/password', [StudentController::class, 'updatePassword']);
```

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 3 minutes
Comment/s: 2

**Comment by catampongan** [comment]
## Approved PR #90  
- [x] Design is the same with the figma design
- [x] User can change their password
- [x] Error handling is present
----------------------------
task-342
task-342-2

------------------------
- Total Test Time: 0.25 hr
- Total Iterations: 0
- Total Comments: 0

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**